packages/arb-token-bridge-ui/src/components/TokenModal/TokenModal.tsx

```diff
@@ -116,10 +161,84 @@ const TokenRow = ({
   )
 }
 
+export const TokenListBody = () => {
+  const {
+    app: {
+      l2NetworkDetails,
+      arbTokenBridge: {
+        bridgeTokens,
+        token: { removeTokensFromList, addTokensFromList }
+      }
+    }
+  } = useAppState()
+  const listsToShow: BridgeTokenList[] = BRIDGE_TOKEN_LISTS.filter(
+    tokenList => {
+      return !!(
+        l2NetworkDetails && tokenList.originChainID === l2NetworkDetails.chainID
+      )
+    }
+  )
+  const toggleTokenList = (
+    bridgeTokenList: BridgeTokenList,
+    isActive: boolean
+  ) => {
+    if (isActive) {
+      removeTokensFromList(bridgeTokenList.id)
+    } else {
+      axios
+        .get(bridgeTokenList.url, {
+          headers: {
+            'Access-Control-Allow-Origin': '*'
+          }
+        })
+        .then(response => {
+          return response.data
+        })
+        .then(tokenListData => {
+          addTokensFromList(tokenListData, bridgeTokenList.id)
```

Comment:
should we add a typeguard here to be sure its correct? could use the uniswap token list validation stuff